test/sql/json/scalar/test_json_extract.test

```diff
@@ -314,3 +314,19 @@ query T
 execute q1('a')
 ----
 1
+
+# test issue 11997
+query I
+select json_extract_string(json('{"j[so]n_\"key": 67}'), '$."j[so]n_\"key"');
```

Comment:
* Can we also add a test with an escaped backslash (`\\`)?
* What if the backslash does not precede another backslash or quote (e.g. `\x`)?

Reply:
I've added tests for these cases, thanks for the suggestion!